api-spec/filters.md

```diff
@@ -0,0 +1,133 @@
+should match all filters to be returned when querying. This implies a logical AND operation. If
+an OR operation is needed, it should be specified through an extension filter.
+
+For the spatial query, we are starting with the simplest form for a client to generate: a bbox. The bbox query 
+is a simple array of WGS84 Longitude and Latitude values. They are in this order: [west, south, east, north]. 
+This query will perform an intersects operation on the geometry values of the items in the catalog. Some GeoJSON 
+objects may provide a bbox property in addition to geometry, but it should not be used for the bbox filter since
+it is an optional field in GeoJSON.
+
+example
+```
+{
+  "bbox": [-180,-90,180,90]
+}
+```
+
+The temporal query will be based on ISO 8601 and should support time ranges as well as equality. To support range
+queries, we are using a simple JSON based language. Ranges will be specified as an object with keys indicating the comparison to use.
+
+Equality is specified as `{"time": "2018-03-20T16:11:44.353Z"}`  
+Before is `{"time":{"lt":"2018-03-20T16:11:44.353Z"}}`  
+After is `{"time":{"gt":"2018-03-20T16:11:44.353Z"}}`  
+Before with Equality is `{"time":{"lte":"2018-03-20T16:11:44.353Z"}}`  
+After with Equality is `{"time":{"gte":"2018-03-20T16:11:44.353Z"}}`  
+
+Filter Extensions
+-----------------
+
+Spatial and Temporal filtering is not enough for many use cases. STAC is designed to be extensible, so we encourage additional filters to be provided as extensions.
+
+Filter extensions are intended to provide a place to specify additional filter types and behavior. Some filters may need to target vendor specific fields or use a more complex structure when defining the filter. For example, a geospatial intersection filter that supports a polygon as input.
+
+Filter extensions should be specified as an openapi fragment. This is a section of YAML that can be incorporated into the larger api definition. These fragments can then be combined into an implementation specific api definition that describes the full api.
+
+Example fragment:
+```
+  IntersectsFilter:
+    type: object
+    description: >-
+          Only returns items that intersect with the provided polygon.
+    properties:
+      intersects:
+        $ref: '#/definitions/Polygon'
+  Polygon:
+    type: object
+    properties:
+      type:
+        type: string
+        enum:
+          - polygon
+      coordinates:
+        $ref: '#/definitions/Polygon2D'
+    required:
+      - type
+      - coordinates
+  Polygon2D:
+    type: array
+    minItems: 1
+    items:
+      $ref: '#/definitions/LinearRing2D'
+  LinearRing2D:
+    type: array
+    minItems: 4
+    items:
+      $ref: '#/definitions/Point2D'
+    example: [[-104,35],[-103,35],[-103,34],[-104,34],[-104,35]]
+  Point2D:
+    description: A 2d geojson point
+    type: array
+    minimum: 2
+    maximum: 2
+    items:
+      type: number
+      format: double
+    example:
+      - -104
+      - 35
+```
```

Comment:
I think it'd be good to publish this as a full 'stac official extension' or something - that we say this is the way to do things. We can debate on if it should be fully core or not, but I think we should have a set of filters that we say are the core extensions, that are recommended for most API's (but we tell clients they need to use the openapi doc to determine what they can query).

Reply:
I agree, I think there has been enough discussion around geo intersects with a polygon that we should at least make it a first tier extension that is "official". Or even possibly part of the core.